### ISSUE TITLE
graylog_event_definition resource should support "key_spec"

### ISSUE THREAD

**Issue by Fullplate** [suzuki-shunsuke/go-graylog#247]
## API Endpoints

POST `/events/definitions` etc.

--

This field is commented out in the [schema definition](https://github.com/suzuki-shunsuke/go-graylog/blob/44cba9b1f1b5030905e22b010e160989f377ffdf/graylog/terraform/resource_event_definition.go#L96) - unsure why.

**Comment by suzuki-shunsuke** [comment]
Thank you for your feedback.

I guess I didn't support key_spec because I didn't know the data structure of `key_spec`.

https://github.com/suzuki-shunsuke/go-graylog/blob/44cba9b1f1b5030905e22b010e160989f377ffdf/graylog/graylog/event_definition.go#L12

If the data structure becomes clear, maybe we can support `key_spec`.

**Comment by Fullplate** [comment]
Thanks for the reply!

The data structure looks simple - an array of event field names.

e.g.

Given event fields with "Is Key?" true:
```
"field_spec": {
    "event_id": {
      "data_type": "string",
      "providers": [
        {
          "type": "template-v1",
          "template": "${source.event_id}",
          "require_values": false
        }
      ]
    },
    "another_field": {
      "data_type": "string",
      "providers": [
        {
          "type": "template-v1",
          "template": "${source.another_field}",
          "require_values": false
        }
      ]
    }
  }
```

`key_spec` will look like:
```
"key_spec": [
    "event_id",
    "another_field"
  ]
```

**Comment by suzuki-shunsuke** [comment]
Thank you for your information!
I have created a pull request. https://github.com/suzuki-shunsuke/go-graylog/pull/248

**Comment by suzuki-shunsuke** [comment]
I have released the pre release version [v11.2.0-0](https://github.com/suzuki-shunsuke/go-graylog/releases/tag/v11.2.0-0).
Please check.
If there is no problem, I'll release v11.2.0 .

**Comment by Fullplate** [comment]
Confirmed working, thank you!

eg

```
field_spec = <<EOF
{
  "event_id": {
    "data_type": "string",
    "providers": [
      {
        "type": "template-v1",
        "template": "$${source.event_id}",
        "require_values": false
      }
    ]
  }
}
EOF

key_spec = ["event_id"]
```

**Comment by suzuki-shunsuke** [comment]
Thank you for your confirming.
I have released [v11.2.0](https://github.com/suzuki-shunsuke/go-graylog/releases/tag/v11.2.0).
